UsersApp/View Controllers/UserDetailsViewController.swift

```diff
@@ -0,0 +1,226 @@
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        
+        if let date = dateFormatter.date(from: dateString) {
+            dateFormatter.dateFormat = "MMM d yyyy, h:m a"
+            return dateFormatter.string(from: date)
+        } else {
+            return "0--0 000"
+        }
+    }
+}
+    
+// MARK: - noteTextViewMethods
+extension UserDetailsViewController {
+    
+    private func configureNoteTextView() {
+        displayTextViewOutline()
+        disableEmptyNoteButtons()
+        
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
+        view.addGestureRecognizer(tapGesture)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(notification: )), name: UIResponder.keyboardWillShowNotification, object: nil)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(notification: )), name: UIResponder.keyboardWillHideNotification, object: nil)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(notification: )), name: UIResponder.keyboardWillChangeFrameNotification, object: self)
+    }
+    
+    @objc private func handleTap(_ gesture: UITapGestureRecognizer) {
+        noteTextView.resignFirstResponder()
+    }
+    
+    @objc private func keyboardWillShow(notification: NSNotification) {
+        guard let keyboardRectangle = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue else {
+            return
+        }
+        
+        if notification.name == UIResponder.keyboardWillShowNotification ||
+        notification.name == UIResponder.keyboardWillChangeFrameNotification {
+            
+            view.frame.origin.y = -keyboardRectangle.height
+        } else {
+            view.frame.origin.y = 0
+        }
+    }
+    
+    private func getUsersEmail() -> String {
+        guard let user = user else {
+            print("error at getting the user's email")
+            return ""
+        }
+        return user.email
+    }
+    
+    private func saveNoteContent() {
+        let email = getUsersEmail()
+        if let previousText = userDefaults.value(forKey: email) as? String {
+            noteTextView.text = previousText
+        }
+    }
+    
+    private func displayTextViewOutline() {
+        noteTextView.layer.borderWidth = 0.7
+        noteTextView.layer.borderColor = UIColor.black.cgColor
+    }
+    
+    internal func textViewDidChange(_ textView: UITextView) {
+        saveButton.isHidden = textView.text.isEmpty
+    }
+    
+    private func disableEmptyNoteButtons() {
+        if noteTextView.text.isEmpty {
+            deleteButton.isHidden = true
+            saveButton.isHidden = true
+        }
+        else {
+            saveButton.isHidden = true
```

Comment:
`saveButton.isHidden = true` is for both is-else cases. take it out of the statement